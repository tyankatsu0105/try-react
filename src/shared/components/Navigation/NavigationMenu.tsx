import React, { FunctionComponent } from 'react';
import { Link as RouterLink } from 'react-router-dom';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';

import {
  List,
  ListItem,
  ListItemIcon,
  ListItemText,
  IconButton,
  Link,
} from '@material-ui/core';
import InboxIcon from '@material-ui/icons/MoveToInbox';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';

const menus = [
  {
    path: '/',
    text: 'ホーム',
  },
  {
    path: '/about',
    text: '自己紹介',
  },
  {
    path: '/material',
    text: 'マテリアル',
  },
  {
    path: '/formik-form',
    text: 'formikフォーム',
  },
  {
    path: '/github-query',
    text: 'GitHub Query',
  },
];

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    'navigationMenu-Header': {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'flex-end',
      padding: `0 ${theme.spacing(1)}`,
      ...theme.mixins.toolbar,
    },
  })
);

interface Props {
  handleDrawerClose: () => void;
}

export const NavigationMenu: FunctionComponent<Props> = (props) => {
  const classes = useStyles();
  const { handleDrawerClose } = props;

  return (
    <nav className="navigationMenu">
      <div className={classes['navigationMenu-Header']}>
        <IconButton onClick={handleDrawerClose}>
          <ChevronLeftIcon />
        </IconButton>
      </div>
      <List className="navigationMenu-List">
        {menus.map((menu) => (
          <Link
            key={menu.path}
            component={RouterLink}
            to={menu.path}
            className="navigationMenu-ListItem"
          >
            <ListItem button>
              <ListItemIcon>
                <InboxIcon />
              </ListItemIcon>
              <ListItemText primary={menu.text} />
            </ListItem>
          </Link>
        ))}
      </List>
    </nav>
  );
};
