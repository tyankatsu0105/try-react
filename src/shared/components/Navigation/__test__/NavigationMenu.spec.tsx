import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';

import { NavigationMenu } from '../NavigationMenu';

describe('NavigationMenu', () => {
  it('snapshot', () => {
    const props = {
      open: true,
      handleDrawerClose: jest.fn(),
    };
    const { baseElement } = render(
      <MemoryRouter>
        <NavigationMenu {...props} />
      </MemoryRouter>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
