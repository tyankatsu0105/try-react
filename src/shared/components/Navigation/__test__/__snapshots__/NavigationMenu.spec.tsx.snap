// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationMenu snapshot 1`] = `
<body>
  <div>
    <nav
      class="navigationMenu"
    >
      <div
        class="makeStyles-navigationMenu-Header-1"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <ul
        class="MuiList-root navigationMenu-List MuiList-padding"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover navigationMenu-ListItem MuiTypography-colorPrimary"
          href="/"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                />
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                ホーム
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </a>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover navigationMenu-ListItem MuiTypography-colorPrimary"
          href="/about"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                />
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                自己紹介
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </a>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover navigationMenu-ListItem MuiTypography-colorPrimary"
          href="/material"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                />
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                マテリアル
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </a>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover navigationMenu-ListItem MuiTypography-colorPrimary"
          href="/formik-form"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                />
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                formikフォーム
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </a>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover navigationMenu-ListItem MuiTypography-colorPrimary"
          href="/github-query"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                />
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
              >
                GitHub Query
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </a>
      </ul>
    </nav>
  </div>
</body>
`;
